eth2near/eth2near-block-relay-rs/src/eth2near_relay.rs

```diff
@@ -62,254 +88,565 @@ impl Eth2NearRelay {
-                }
-                Err(err) => {
-                    warn!(target: "relay", "Error \"{}\" on getting light client update. Skipping sending light client update", err)
-                }
+        println!("Submitted blocks: {}", blocks.len());
+
+        relay
+            .eth_client_contract
+            .send_headers(&blocks, end_slot)
+            .unwrap();
+    }
+
+    fn send_blocks_till_finalized_eth_slot(relay: &mut Eth2NearRelay, finality_slot: u64) -> u64 {
+        let mut slot = finality_slot + 1;
+
+        let mut finality_slot_on_eth = relay
+            .beacon_rpc_client
+            .get_last_finalized_slot_number()
+            .unwrap()
+            .as_u64();
+
+        let mut blocks: Vec<BlockHeader> = vec![];
+        while finality_slot == finality_slot_on_eth || slot <= finality_slot_on_eth {
+            if let Ok(block) = relay.get_execution_block_by_slot(slot) {
+                blocks.push(block)
             }
-        } else {
-            debug!(target: "relay", "Finalized period on Eth and Near are different. Fetching sync commity update");
-            match self
+            slot += 1;
+
+            finality_slot_on_eth = relay
                 .beacon_rpc_client
-                .get_finality_light_client_update_with_sync_commity_update()
-            {
-                Ok(light_client_update) => {
-                    self.send_specific_light_cleint_update(light_client_update)
-                }
-                Err(err) => {
-                    warn!(target: "relay", "Error \"{}\" on getting light client update. Skipping sending light client update", err)
-                }
-            }
+                .get_last_finalized_slot_number()
+                .unwrap()
+                .as_u64();
         }
+
+        relay
+            .eth_client_contract
+            .send_headers(&blocks, finality_slot_on_eth)
+            .unwrap();
+
+        finality_slot_on_eth
     }
 
-    fn send_hand_made_light_client_update(
-        &mut self,
-        last_finalized_slot_on_near: u64,
-        last_submitted_slot: u64,
-    ) {
-        trace!(target: "relay", "last_finalized_slot_on_near {}", last_finalized_slot_on_near);
-        if (last_submitted_slot as i64) - (last_finalized_slot_on_near as i64)
-            < (self.current_gap_between_finalized_and_attested_slot as i64)
-        {
-            info!(target: "relay", "Waiting for sending more headers to near. Skip sending light client update.");
-            return;
+    fn get_finalized_slot(relay: &Eth2NearRelay) -> u64 {
+        relay
+            .eth_client_contract
+            .get_finalized_beacon_block_slot()
+            .unwrap()
+    }
+
+    #[test]
+    fn test_send_specific_light_client_update() {
+        let mut relay = get_relay(true, true);
+        let mut slot = get_finalized_slot(&relay);
+        slot += 1;
+
+        send_execution_blocks(&mut relay, slot, 1099392);
```

Comment:
I think it's better to use some constant and/or calculation for the last block. Otherwise, it looks strange, we retrieve the finalized slot from the function, but the end block height is hardcoded.

Reply:
fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/787/commits/e83a9b8c2fa5c7157e8813ee300566e4ca48fa14